util/topsql/reporter/datamodel.go

```diff
@@ -378,11 +379,20 @@ func (r *record) rebuildTsIndex() {
 	}
 }
 
+func rawDigestToSlice(s parser.RawDigestString) []byte {
+	// TODO: maybe it's fine to be just `[]byte{}`.
+	if s == "" {
+		return nil
+	}
+	return []byte(s)
```

Comment:
Does `[]byte(s)` alloc?

Reply:
Yes, it does. The reason has been explained at the call site: We cannot ensure `tipb.TopSQLRecord` is not modified (although from the current code base it has not been modified).

// Do slice copy, since we don't know whether tipb.TopSQLRecord will be modified.